Adding process related information

rahuyada opened the pull request:
1: Created files for process related information
2: Added a new resource id for process information.
3: Added code to get process information.

brauner commented on resource.c:
```diff
@@ -217,6 +234,7 @@ int res_read_blk(res_blk_t *res, int pid, int flags)
 {
 	int ismeminforeq = 0;
 	int isnetdevreq = 0;
+	int isprocreq = 0;
```
Hm, just realizing this now. Wouldn't it be cleaner to use proper booleans for this?

rahuyada replied:
agree.

brauner commented on resource.h:
```diff
@@ -143,6 +150,55 @@ typedef struct res_net_ifstat {
+	unsigned long lib;
+	unsigned long data;
+	unsigned long dt;
+} res_proc_infoall_t;
```
Would it make sense to be even more specific about types? So instead of e.g. `unsigned long long` just use `uint64_t`, in fact you could probably use `uint64_t` also for `unsigned long`.

rahuyada replied:
makes sense, will change here and some other places.

brauner commented on resproc.c:
```diff
@@ -0,0 +1,195 @@
+	tmp = strrchr(buf, ')');
+	num = tmp - buf;
+	if(num >= sizeof p->cmd)
+		num = sizeof p->cmd - 1;
```
Hm, are you using `sizeof` without `()` everywhere? For consistency you should probably pick one of the two `sizeof` or `sizeof()`. So far we have always used the latter.

brauner commented on resproc.c:
```diff
@@ -0,0 +1,195 @@
+		&p->rtprio, &p->policy
+	);
+
+	if(!p->nlwp){
```
Missing space between `if` and `(`.

brauner commented on resproc.c:
```diff
@@ -0,0 +1,195 @@
+	/* Get user id and use that to get useranme */
+	if (stat(path, &sb) == -1) {
+		eprintf("Error in reading stat");
+	}
```
Hm, we usually don't do `{}` around single-line if statements but up to you. Just keep it consistent across the codebase. :)

brauner commented on resproc.c:
```diff
@@ -0,0 +1,195 @@
+		sprintf(p->euser, "%u", p->euid);
+	} else {
+		strcpy(p->euser, pw->pw_name);
+	}
```
Hm, we usually don't do `{}` around single-line if statements but up to you. Just keep it consistent across the codebase. :)

rahuyada replied:
got it.

brauner commented on resproc.c:
```diff
@@ -0,0 +1,195 @@
+	if(!pw || strlen(pw->pw_name) >= RES_USR_NAME_SZ) {
+		sprintf(p->euser, "%u", p->euid);
+	} else {
+		strcpy(p->euser, pw->pw_name);
```
I'm not a fan of using `strcpy` at all tbh.

rahuyada replied:
strncpy maybe ?

brauner replied:
Let's leave it for now. In LXC we'll carry a custom very dumb strlcpy() :)

brauner commented on resproc.c:
```diff
@@ -0,0 +1,195 @@
+	p->euid = sb.st_uid;
+	pw = getpwuid(p->euid);
+	if(!pw || strlen(pw->pw_name) >= RES_USR_NAME_SZ) {
+		sprintf(p->euser, "%u", p->euid);
```
Hm, `snprintf` maybe?

rahuyada replied:
Done.

brauner commented on resproc.c:
```diff
@@ -0,0 +1,195 @@
+	res_proc_infoall_t *p;
+	char buf[PROCBUF_1024];
+
+#define CHECK_SIZE(sz, req_sz)						\
```
Why the define if you just use this code in a single place? :)

rahuyada replied:
It will come handy when we add more resources to resproc.c

rahuyada commented:
I have made code changes based on comments provided by @brauner.

brauner commented:
@hallyn, any comments? :)

rahuyada commented:
@hallyn @brauner Please let me know if there are any more comments. If things look good please merge it.

hallyn commented:
(sorry, looking)

hallyn commented:
Were you planning on squashing these?  i assume there aren't other branches to cherrypick to so in general keeping them split would be fine, except the split between patches 4 and 5 regarding PROC_MAX is awkward - RES_PROC_PPID is removed in patch 4, and the later ones are renumbered only in the otherwise-unrelated patch 5.

hallyn commented:
LGTM, thanks.

rahuyada commented:
> Were you planning on squashing these? i assume there aren't other branches to cherrypick to so in general keeping them split would be fine, except the split between patches 4 and 5 regarding PROC_MAX is awkward - RES_PROC_PPID is removed in patch 4, and the later ones are renumbered only in the otherwise-unrelated patch 5.

I had added RES_PROC_PPID thinking each proc related info can be provided one by one, however later I decided to club all proc related info in one struct and assigned res_id RES_PROC_INFOALL to it. However I forgot to delete RES_PROC_PPID, so i deleted that in patch 4. I wanted it to be like this - 
#define PROC_MIN			4096
#define RES_PROC_INFOALL	4097
#define PROC_MAX			4098
But again I realized I had not numbered PROC_INFOALL and PROC_MAX correctly. 
In theory it is ok to keep PROC_MIN and PROC_MAX anything as far as all proc res_ids are between them. But since the changes were not merged yet, I fixed them in last patch set.
I will be more careful in future.

hallyn commented:
> > Were you planning on squashing these? i assume there aren't other branches to cherrypick to so in general keeping them split would be fine, except the split between patches 4 and 5 regarding PROC_MAX is awkward - RES_PROC_PPID is removed in patch 4, and the later ones are renumbered only in the otherwise-unrelated patch 5.
> 
> I had added RES_PROC_PPID thinking each proc related info can be provided one by one, however later I decided to club all proc related info in one struct and assigned res_id RES_PROC_INFOALL to it. However I forgot to delete RES_PROC_PPID, so i deleted that in patch 4. I wanted it to be like this -
> #define PROC_MIN 4096
> #define RES_PROC_INFOALL 4097
> #define PROC_MAX 4098
> But again I realized I had not numbered PROC_INFOALL and PROC_MAX correctly.
> In theory it is ok to keep PROC_MIN and PROC_MAX anything as far as all proc res_ids are between them. But since the changes were not merged yet, I fixed them in last patch set.
> I will be more careful in future.

That's all fine - but is there any reason why you can't squash these commits now?  i.e.

```
git rebase -i origin/master
# set all but the first to 's'
git push -force origing
```

?

rahuyada commented:
> > > Were you planning on squashing these? i assume there aren't other branches to cherrypick to so in general keeping them split would be fine, except the split between patches 4 and 5 regarding PROC_MAX is awkward - RES_PROC_PPID is removed in patch 4, and the later ones are renumbered only in the otherwise-unrelated patch 5.
> > 
> > 
> > I had added RES_PROC_PPID thinking each proc related info can be provided one by one, however later I decided to club all proc related info in one struct and assigned res_id RES_PROC_INFOALL to it. However I forgot to delete RES_PROC_PPID, so i deleted that in patch 4. I wanted it to be like this -
> > #define PROC_MIN 4096
> > #define RES_PROC_INFOALL 4097
> > #define PROC_MAX 4098
> > But again I realized I had not numbered PROC_INFOALL and PROC_MAX correctly.
> > In theory it is ok to keep PROC_MIN and PROC_MAX anything as far as all proc res_ids are between them. But since the changes were not merged yet, I fixed them in last patch set.
> > I will be more careful in future.
> 
> That's all fine - but is there any reason why you can't squash these commits now? i.e.
> 
> ```
> git rebase -i origin/master
> # set all but the first to 's'
> git push -force origing
> ```
> 
> ?

Ok, I get it now :).
Yes, I have just squashed them in one commit.

hallyn commented:
Thanks :)
